File: 0050-powx-n/0050-powx-n.cpp
class Solution {
public:
    double myPow(double x, int k) {
        bool nflag=false;
        double sol=1;
        if(k<0)nflag=true;
        long n=abs(k);
        while(n>0)
            if(n%2)sol*=x,n--;
            else x*=x,n/=2;
        return nflag?1.0/sol:sol;
        
    }
};

File: 0050-powx-n/0050-powx-n.java
class Solution {
    public double myPow(double x, int n) {
      long nn=n;
      double ans=1.0;
      boolean flag =false;
      if(nn<0)
      {
        flag =true;
        nn*=-1;
      } 

     while(nn>0)
     {
       if(nn%2==1)
       {
        ans=ans*x;
        nn--;
       }
       else
       {
        x=x*x;
        nn=nn/2;
       }
     }
     if(flag)return 1.0/ans;
     return ans;


    }
}